File: ts-sdk/src/sdk/displayads.ts
import { AxiosInstance, AxiosRequestConfig, AxiosResponse } from "axios";
import * as operations from "./models/operations";
import * as utils from "../internal/utils";

export class DisplayAds {
  _defaultClient: AxiosInstance;
  _securityClient: AxiosInstance;
  _serverURL: string;
  _language: string;
  _sdkVersion: string;
  _genVersion: string;

  constructor(defaultClient: AxiosInstance, securityClient: AxiosInstance, serverURL: string, language: string, sdkVersion: string, genVersion: string) {
    this._defaultClient = defaultClient;
    this._securityClient = securityClient;
    this._serverURL = serverURL;
    this._language = language;
    this._sdkVersion = sdkVersion;
    this._genVersion = genVersion;
  }
  
  /**
   * getApiDisplayAds - display ads
   *
   * This endpoint allows the client to retrieve a list of all display ads.
  **/
  getApiDisplayAds(
    config?: AxiosRequestConfig
  ): Promise<operations.GetApiDisplayAdsResponse> {
    const baseURL: string = this._serverURL;
    const url: string = baseURL.replace(/\/$/, "") + "/api/display_ads";
    
    const client: AxiosInstance = this._securityClient!;
    
    
    const r = client.request({
      url: url,
      method: "get",
      ...config,
    });
    
    return r.then((httpRes: AxiosResponse) => {
        const contentType: string = httpRes?.headers?.["content-type"] ?? "";

        if (httpRes?.status == null) throw new Error(`status code not found in response: ${httpRes}`);
        const res: operations.GetApiDisplayAdsResponse = {statusCode: httpRes.status, contentType: contentType};
        switch (true) {
          case httpRes?.status == 200:
            break;
          case httpRes?.status == 401:
            break;
        }

        return res;
      })
  }

  
  /**
   * getApiDisplayAdsId - display ad
   *
   * This endpoint allows the client to retrieve a single display ad, via its id.
  **/
  getApiDisplayAdsId(
    req: operations.GetApiDisplayAdsIdRequest,
    config?: AxiosRequestConfig
  ): Promise<operations.GetApiDisplayAdsIdResponse> {
    if (!(req instanceof utils.SpeakeasyBase)) {
      req = new operations.GetApiDisplayAdsIdRequest(req);
    }
    
    const baseURL: string = this._serverURL;
    const url: string = utils.generateURL(baseURL, "/api/display_ads/{id}", req.pathParams);
    
    const client: AxiosInstance = this._securityClient!;
    
    
    const r = client.request({
      url: url,
      method: "get",
      ...config,
    });
    
    return r.then((httpRes: AxiosResponse) => {
        const contentType: string = httpRes?.headers?.["content-type"] ?? "";

        if (httpRes?.status == null) throw new Error(`status code not found in response: ${httpRes}`);
        const res: operations.GetApiDisplayAdsIdResponse = {statusCode: httpRes.status, contentType: contentType};
        switch (true) {
          case httpRes?.status == 200:
            break;
          case httpRes?.status == 401:
            break;
          case httpRes?.status == 404:
            break;
        }

        return res;
      })
  }

  
  /**
   * postApiDisplayAds - display ads
   *
   * This endpoint allows the client to create a new display ad.
  **/
  postApiDisplayAds(
    req: operations.PostApiDisplayAdsRequest,
    config?: AxiosRequestConfig
  ): Promise<operations.PostApiDisplayAdsResponse> {
    if (!(req instanceof utils.SpeakeasyBase)) {
      req = new operations.PostApiDisplayAdsRequest(req);
    }
    
    const baseURL: string = this._serverURL;
    const url: string = baseURL.replace(/\/$/, "") + "/api/display_ads";

    let [reqBodyHeaders, reqBody]: [object, any] = [{}, {}];

    try {
      [reqBodyHeaders, reqBody] = utils.serializeRequestBody(req);
    } catch (e: unknown) {
      if (e instanceof Error) {
        throw new Error(`Error serializing request body, cause: ${e.message}`);
      }
    }
    
    const client: AxiosInstance = this._securityClient!;
    
    const headers = {...reqBodyHeaders, ...config?.headers};
    
    const r = client.request({
      url: url,
      method: "post",
      headers: headers,
      data: reqBody, 
      ...config,
    });
    
    return r.then((httpRes: AxiosResponse) => {
        const contentType: string = httpRes?.headers?.["content-type"] ?? "";

        if (httpRes?.status == null) throw new Error(`status code not found in response: ${httpRes}`);
        const res: operations.PostApiDisplayAdsResponse = {statusCode: httpRes.status, contentType: contentType};
        switch (true) {
          case httpRes?.status == 200:
            break;
          case httpRes?.status == 401:
            break;
          case httpRes?.status == 422:
            break;
        }

        return res;
      })
  }

  
  /**
   * putApiDisplayAdsId - display ads
   *
   * This endpoint allows the client to update the attributes of a single display ad, via its id.
  **/
  putApiDisplayAdsId(
    req: operations.PutApiDisplayAdsIdRequest,
    config?: AxiosRequestConfig
  ): Promise<operations.PutApiDisplayAdsIdResponse> {
    if (!(req instanceof utils.SpeakeasyBase)) {
      req = new operations.PutApiDisplayAdsIdRequest(req);
    }
    
    const baseURL: string = this._serverURL;
    const url: string = utils.generateURL(baseURL, "/api/display_ads/{id}", req.pathParams);

    let [reqBodyHeaders, reqBody]: [object, any] = [{}, {}];

    try {
      [reqBodyHeaders, reqBody] = utils.serializeRequestBody(req);
    } catch (e: unknown) {
      if (e instanceof Error) {
        throw new Error(`Error serializing request body, cause: ${e.message}`);
      }
    }
    
    const client: AxiosInstance = this._securityClient!;
    
    const headers = {...reqBodyHeaders, ...config?.headers};
    
    const r = client.request({
      url: url,
      method: "put",
      headers: headers,
      data: reqBody, 
      ...config,
    });
    
    return r.then((httpRes: AxiosResponse) => {
        const contentType: string = httpRes?.headers?.["content-type"] ?? "";

        if (httpRes?.status == null) throw new Error(`status code not found in response: ${httpRes}`);
        const res: operations.PutApiDisplayAdsIdResponse = {statusCode: httpRes.status, contentType: contentType};
        switch (true) {
          case httpRes?.status == 200:
            break;
          case httpRes?.status == 401:
            break;
          case httpRes?.status == 404:
            break;
        }

        return res;
      })
  }

  
  /**
   * putApiDisplayAdsIdUnpublish - unpublish
   *
   * This endpoint allows the client to remove a display ad from rotation by un-publishing it.
  **/
  putApiDisplayAdsIdUnpublish(
    req: operations.PutApiDisplayAdsIdUnpublishRequest,
    config?: AxiosRequestConfig
  ): Promise<operations.PutApiDisplayAdsIdUnpublishResponse> {
    if (!(req instanceof utils.SpeakeasyBase)) {
      req = new operations.PutApiDisplayAdsIdUnpublishRequest(req);
    }
    
    const baseURL: string = this._serverURL;
    const url: string = utils.generateURL(baseURL, "/api/display_ads/{id}/unpublish", req.pathParams);
    
    const client: AxiosInstance = this._securityClient!;
    
    
    const r = client.request({
      url: url,
      method: "put",
      ...config,
    });
    
    return r.then((httpRes: AxiosResponse) => {
        const contentType: string = httpRes?.headers?.["content-type"] ?? "";

        if (httpRes?.status == null) throw new Error(`status code not found in response: ${httpRes}`);
        const res: operations.PutApiDisplayAdsIdUnpublishResponse = {statusCode: httpRes.status, contentType: contentType};
        switch (true) {
          case httpRes?.status == 204:
            break;
          case httpRes?.status == 401:
            break;
          case httpRes?.status == 404:
            break;
        }

        return res;
      })
  }

}

File: python-sdk/src/forem/display_ads.py
import requests
from forem.models import operations
from . import utils

class DisplayAds:
    _client: requests.Session
    _security_client: requests.Session
    _server_url: str
    _language: str
    _sdk_version: str
    _gen_version: str

    def __init__(self, client: requests.Session, security_client: requests.Session, server_url: str, language: str, sdk_version: str, gen_version: str) -> None:
        self._client = client
        self._security_client = security_client
        self._server_url = server_url
        self._language = language
        self._sdk_version = sdk_version
        self._gen_version = gen_version

    
    def get_api_display_ads(self) -> operations.GetAPIDisplayAdsResponse:
        r"""display ads
        This endpoint allows the client to retrieve a list of all display ads.
        """
        
        base_url = self._server_url
        
        url = base_url.removesuffix("/") + "/api/display_ads"
        
        
        client = self._security_client
        
        r = client.request("GET", url)
        content_type = r.headers.get("Content-Type")

        res = operations.GetAPIDisplayAdsResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            pass
        elif r.status_code == 401:
            pass

        return res

    
    def get_api_display_ads_id_(self, request: operations.GetAPIDisplayAdsIDRequest) -> operations.GetAPIDisplayAdsIDResponse:
        r"""display ad
        This endpoint allows the client to retrieve a single display ad, via its id.
        """
        
        base_url = self._server_url
        
        url = utils.generate_url(base_url, "/api/display_ads/{id}", request.path_params)
        
        
        client = self._security_client
        
        r = client.request("GET", url)
        content_type = r.headers.get("Content-Type")

        res = operations.GetAPIDisplayAdsIDResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            pass
        elif r.status_code == 401:
            pass
        elif r.status_code == 404:
            pass

        return res

    
    def post_api_display_ads(self, request: operations.PostAPIDisplayAdsRequest) -> operations.PostAPIDisplayAdsResponse:
        r"""display ads
        This endpoint allows the client to create a new display ad.
        """
        
        base_url = self._server_url
        
        url = base_url.removesuffix("/") + "/api/display_ads"
        
        headers = {}
        req_content_type, data, json, files = utils.serialize_request_body(request)
        if req_content_type != "multipart/form-data" and req_content_type != "multipart/mixed":
            headers["content-type"] = req_content_type
        
        client = self._security_client
        
        r = client.request("POST", url, data=data, json=json, files=files, headers=headers)
        content_type = r.headers.get("Content-Type")

        res = operations.PostAPIDisplayAdsResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            pass
        elif r.status_code == 401:
            pass
        elif r.status_code == 422:
            pass

        return res

    
    def put_api_display_ads_id_(self, request: operations.PutAPIDisplayAdsIDRequest) -> operations.PutAPIDisplayAdsIDResponse:
        r"""display ads
        This endpoint allows the client to update the attributes of a single display ad, via its id.
        """
        
        base_url = self._server_url
        
        url = utils.generate_url(base_url, "/api/display_ads/{id}", request.path_params)
        
        headers = {}
        req_content_type, data, json, files = utils.serialize_request_body(request)
        if req_content_type != "multipart/form-data" and req_content_type != "multipart/mixed":
            headers["content-type"] = req_content_type
        
        client = self._security_client
        
        r = client.request("PUT", url, data=data, json=json, files=files, headers=headers)
        content_type = r.headers.get("Content-Type")

        res = operations.PutAPIDisplayAdsIDResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            pass
        elif r.status_code == 401:
            pass
        elif r.status_code == 404:
            pass

        return res

    
    def put_api_display_ads_id_unpublish(self, request: operations.PutAPIDisplayAdsIDUnpublishRequest) -> operations.PutAPIDisplayAdsIDUnpublishResponse:
        r"""unpublish
        This endpoint allows the client to remove a display ad from rotation by un-publishing it.
        """
        
        base_url = self._server_url
        
        url = utils.generate_url(base_url, "/api/display_ads/{id}/unpublish", request.path_params)
        
        
        client = self._security_client
        
        r = client.request("PUT", url)
        content_type = r.headers.get("Content-Type")

        res = operations.PutAPIDisplayAdsIDUnpublishResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 204:
            pass
        elif r.status_code == 401:
            pass
        elif r.status_code == 404:
            pass

        return res

    